Program.cs

```diff
@@ -0,0 +1,127 @@
+{
+    switch (inputNumber)
+    {
+        case 1:
+            Console.WriteLine("Enter 1. to create a Square");
+            Console.WriteLine("Enter 2. to create a Rectangle");
+            Console.WriteLine("Enter 3. to create a Rhombus");
+            Console.WriteLine("Enter 4. to create a Trapeze");
+            Console.WriteLine("Enter 5. to create a Triangle");
+            int shapeChoice = Convert.ToInt32(Console.ReadLine());
+            switch (shapeChoice)
+            {
+                case 1:
+                    Console.WriteLine("Enter edge length");
+                    int edgeSqaure = Convert.ToInt32(Console.ReadLine());
+                    Shape squareShape = new Square(edgeSqaure);
+                    shapesList.Add(squareShape);
+                    break;
+                case 2:
+                    Console.WriteLine("Enter first and second edges lengths");
+                    int firstEdgeRect = Convert.ToInt32(Console.ReadLine());
+                    int secondEdgeRect = Convert.ToInt32(Console.ReadLine());
+                    Shape rectangleShape = new Rectangle(firstEdgeRect, secondEdgeRect);
+                    shapesList.Add(rectangleShape);
+                    break;
+                case 3:
+                    Console.WriteLine("Enter edge length");
+                    int edgeRhombus = Convert.ToInt32(Console.ReadLine());
+                    Shape rhombusShape = new Rhombus(edgeRhombus);
+                    shapesList.Add(rhombusShape);
+                    break;
+                case 4:
+                    Console.WriteLine("Enter base length and trapeze height");
+                    int tbase = Convert.ToInt32(Console.ReadLine());
+                    int height = Convert.ToInt32(Console.ReadLine());
+                    Shape trapezeShape = new Trapeze(tbase, height);
+                    shapesList.Add(trapezeShape);
+                    break;
+                case 5:
+                    Console.WriteLine("Enter edge length");
+                    int edgeTriangle = Convert.ToInt32(Console.ReadLine());
+                    Shape triangleShape = new Triangle(edgeTriangle);
+                    shapesList.Add(triangleShape);
+                    break;
+                default:
+                    Console.WriteLine("Cant read " + shapeChoice);
+                    continue;
+            }
+            break;
+        case 2:
+            Helper.printList(shapesList);
+            Console.WriteLine("Choose a shape to update");
+            int updateShape = Convert.ToInt32(Console.ReadLine());
+            switch (shapesList[updateShape - 1].GetType().Name)
+            {
+                case "Square":
+                    Console.WriteLine("Enter Square new edge length");
+                    int squareNewEdge = Convert.ToInt32(Console.ReadLine());
+                    Shape newSquare = new Square(squareNewEdge);
+                    shapesList[updateShape - 1] = newSquare;
```

Comment:
each shape class should contain a functionality of updating. The main class shouldn't be dealing with each of the shape's functionality